internal/receiver/scriptedinputsreceiver/procpipe/commander.go

```diff
@@ -0,0 +1,203 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procpipe
+
+import (
+	"context"
+	_ "embed"
+	"io"
+	"os/exec"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const includePattern = ". \"$(dirname \"$0\")\"/common.sh"
+
+//go:embed scripts/cpu.sh
+var cpuScript string
+
+//go:embed scripts/df.sh
+var dfScript string
+
+//go:embed scripts/ps.sh
+var psScript string
+
+//go:embed scripts/common.sh
+var commonScript string
+
+var scripts = map[string]string{
+	"cpu": cpuScript,
+	"df":  dfScript,
+	"ps":  psScript,
+}
+
+// Commander can start/stop/restart the Agent executable and also watch for a signal
+// for the Agent process to finish.
```

Comment:
not to start a semantic dispute, but I wouldn't consider these scripts "agents".

```suggestion
// Commander can start/stop/restart the shell executable and also watch for a signal
// for the process to finish.
```